src/navigators/createNavigator.js

```diff
@@ -42,6 +42,11 @@ function createNavigator(NavigatorView, router, navigationConfig) {
       const { routes } = state;
 
       const descriptors = {};
+      if (typeof routes === 'undefined') {
+        throw TypeError(
+          'No "routes" found in navigation state. Did you try to pass the navigation prop of a React component to a Navigator child? See https://github.com/react-navigation/react-navigation/issues/3598#issuecomment-375622188'
```

Comment:
rather than referring to a comment in an issue, can we link to the docs instead?

This might be an appropriate location: https://next.reactnavigation.org/docs/custom-navigators.html#navigator-navigation-prop

Or you could improve a different doc, and link to that

Reply:
It looks like this doc might also be relevant: https://v2.reactnavigation.org/docs/common-mistakes.html#explicitly-rendering-more-than-one-navigator

Reply:
Don't forget to use `new TypeError` - useful stack traces depend on the constructor being called.